pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
"stop is closed once Run returns and Run only returns once process returns" while this is technically true, these two things serve different purposes.

The goal of `stop` is if we later add signal catching to main (code change, such as if someone hits ctrl+c). Stopping the channel stops all goroutines or sleeps (if we use sleepOrDone) the agent was doing. The fact that this function returns and Run returns doesn't matter so much as if written for signal catching, the Run is in its own goroutine. Note that its not this way now.

Basically, we should either use an interruptible sleep or (better the more I think about it) decide to just return immedietaly - sleeping seems pointless. Either way, the prior `time.Sleep` should go.